lib/minidoc/timestamps.rb

```diff
@@ -15,6 +15,28 @@ def timestamps!
     end
   end
 
+  def set(attributes)
+    if self.class.record_timestamps
+      super(attributes.merge(updated_at: Time.now.utc))
+    else
+      super(attributes)
+    end
+  end
+
+  def atomic_set(query, attributes)
+    if self.class.record_timestamps
+      super(query, attributes.merge(updated_at: Time.now.utc))
+    else
+      super(query, attributes)
+    end
+  end
+
+  def unset(*keys)
+    super
+
+    set(updated_at: Time.now.utc) if self.class.record_timestamps
```

Comment:
kind of interesting maybe -- this will go through the brand-new `set` method defined on L18, which already sort of does this. The duplication might be OK (the alternative that comes to mind is changing this to a kind of cryptic-looking `set({})`) but maybe worth considering for a moment... :)